ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -389,18 +404,47 @@ func (h *FaultHandler) StartNetworkPacketLoss() func(http.ResponseWriter, *http.
 		rwMu.Lock()
 		defer rwMu.Unlock()
 
-		// TODO: Check status of current fault injection
-		// TODO: Invoke the start fault injection functionality if not running
-
-		responseBody := types.NewNetworkFaultInjectionSuccessResponse("running")
-		logger.Info("Successfully started fault", logger.Fields{
+		var stringToBeLogged string
+		var responseBody types.NetworkFaultInjectionResponse
+		var httpStatusCode int
+		// Check the status of current fault injection.
+		latencyFaultExists, packetLossFaultExists, err := h.checkTCFault(taskMetadata)
```

Comment:
Q: Should we have only one context with a timeout duration which can be shared among exec calls?

Reply:
Good question. I'm not sure about this. Will have to check with Xing to see what's the right thing to do.

Reply:
Agree that we should use one context.

Reply:
Confirmed with Xing, all commands should share the same context. I will make updates to reflect this.